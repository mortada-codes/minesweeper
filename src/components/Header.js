
import React,{useEffect,useState} from 'react';
import { View,Text,StyleSheet,} from 'react-native';

import { formatTime } from '../util';






export default  ({isPlaying})=>{
    const [gameTime,setGameTime] = useState(0)
            useEffect(()=>{
                 if(!isPlaying) return;   
                const INTERVALID =   setInterval(()=>{
                 setGameTime(prv=>++prv)           
                },1000) 
                return ()=> INTERVALID && clearInterval(INTERVALID)   
            },[isPlaying])


    return (<View style={{flexDirection:'row',justifyContent:'space-between'}}>
        <Text style={{fontSize:26}}>00</Text>
           <Text style={{fontSize:26}} >{formatTime(gameTime)}</Text> 
           <Text style={{fontSize:26}}>00</Text>
    </View>)
}
