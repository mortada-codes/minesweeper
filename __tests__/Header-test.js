/**
 * @format
 */

import 'react-native';
import React from 'react';
import Header from '../src/components/Header';

// Note: test renderer must be required after react-native.
import renderer from 'react-test-renderer';
import { fireEvent, render, wait } from '@testing-library/react-native';


 
it('renders correctly', () => {
   const {asJSON} =  render(<Header />);
   expect(asJSON()).toMatchSnapshot();

});
