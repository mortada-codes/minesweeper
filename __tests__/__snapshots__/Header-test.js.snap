// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  collapsable={true}
  pointerEvents="box-none"
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    style={
      Object {
        "flexDirection": "row",
        "justifyContent": "space-between",
      }
    }
  >
    <Text
      style={
        Object {
          "fontSize": 26,
        }
      }
    >
      00
    </Text>
    <Text
      style={
        Object {
          "fontSize": 26,
        }
      }
    >
      00:00:00
    </Text>
    <Text
      style={
        Object {
          "fontSize": 26,
        }
      }
    >
      00
    </Text>
  </View>
</View>
`;
